distribution/src/bin-filemode-755/hz-start

```diff
@@ -34,9 +34,18 @@ fi
 
 . "$SCRIPT_DIR"/common.sh
 
-# HAZELCAST_CONFIG holds path to the configuration file. The path is relative to the Hazelcast installation (HAZELCAST_HOME).
+# HAZELCAST_CONFIG holds the configuration path.
+# If the the path is relative, it is relative to the Hazelcast installation directory (HAZELCAST_HOME).
+# The path can be absolute.
+# By default, it is set to $HAZELCAST_HOME/config/hazelcast.xml
 if [ -z "$HAZELCAST_CONFIG" ]; then
-    HAZELCAST_CONFIG="config/hazelcast.xml"
+    HAZELCAST_CONFIG="$HAZELCAST_HOME/config/hazelcast.xml"
+else
```

Comment:
You can avoid nesting conditions: in the first `if` you can leave the code as it was and simply add another `if` below:
```shell
if [ -z "$HAZELCAST_CONFIG" ]; then
    HAZELCAST_CONFIG="config/hazelcast.xml"
fi

# if the first character is /, then this is an absolute path, use as is
# otherwise prepend (HAZELCAST_HOME)
if [ "${HAZELCAST_CONFIG:0:1}" != "/" ]; then
  HAZELCAST_CONFIG="$HAZELCAST_HOME/$HAZELCAST_CONFIG"
fi
```

Reply:
That makes sense. Updated at: https://github.com/hazelcast/hazelcast/pull/19908/commits/cda27a4a6443a67f6a953083ac58bca784f79662